### PR TITLE
Memory leak in intvector classes fixed

### PR DESCRIPTION
**Pull request by ashermancinelli**
I found this bug reviewing the impl since according to @pelesh we'll be using these in the near future. CC @cnpetra

### CODE REVIEW

**Comment by ashermancinelli** [comment]
@pelesh Can you take a quick look at this?

**Review by pelesh — CHANGES_REQUESTED** [review]
I believe logic for allocating RAJA vector needs to be updated per my earlier comment. Also see [hiopVectorRajaPar](https://github.com/LLNL/hiop/blob/develop/src/LinAlg/hiopVectorRajaPar.cpp#L140) class implementation.

File: src/LinAlg/hiopVectorIntRaja.cpp [hunk 1]
```diff
@@ -80,6 +80,20 @@ hiopVectorIntRaja::hiopVectorIntRaja(int sz, std::string mem_space)
...
+  auto& resmgr = umpire::ResourceManager::getInstance();
+  umpire::Allocator devalloc  = resmgr.getAllocator(mem_space_);
+  devalloc.deallocate(buf_dev_);
+  if (mem_space_ != "HOST")
+  {
+    umpire::Allocator hostalloc = resmgr.getAllocator("HOST");
+    hostalloc.deallocate(buf_host_);
+  }
+  buf_host_ = nullptr;
+  buf_dev_ = nullptr;
```
[the remark this answers is not in the document]
Comment (ashermancinelli) [reply src/LinAlg/hiopVectorIntRaja.cpp:94]: This is a good point - thanks for the review!

[the remark this answers is not in the document]
Comment (ashermancinelli) [reply src/LinAlg/hiopVectorIntRaja.cpp]: I believe this should be resolved in 4203d6d.

**Comment by ashermancinelli** [comment]
Rebased on recent develop. I also ran into a bug with our Valgrind tests in CMake, so there is a fix for that too.

**Review by pelesh — APPROVED** [review]
Looks good! I think we can merge.

**Pull request merged**